pkg/synthesis/k8sPolicies.go

```diff
@@ -0,0 +1,135 @@
+	dfw.ActionDeny:      admin.AdminNetworkPolicyRuleActionDeny,
+	dfw.ActionJumpToApp: admin.AdminNetworkPolicyRuleActionPass,
+}
+
+type k8sPolicies struct {
+	networkPolicies      []*networking.NetworkPolicy
+	adminNetworkPolicies []*admin.AdminNetworkPolicy
+}
+
+func (policies *k8sPolicies) toNetworkPolicies(model *AbstractModelSyn) ([]*networking.NetworkPolicy, []*admin.AdminNetworkPolicy) {
+	for _, p := range model.policy {
+		policies.symbolicRulesToPolicies(model, p.outbound, false)
+		policies.symbolicRulesToPolicies(model, p.inbound, true)
+	}
+	return policies.networkPolicies, policies.adminNetworkPolicies
+}
+
+func (policies *k8sPolicies) symbolicRulesToPolicies(model *AbstractModelSyn, rules []*symbolicRule, inbound bool) {
+	for _, rule := range rules {
+		isAdmin := model.allowOnlyFromCategory > rule.origRuleCategory
+		paths := &rule.allowOnlyRulePaths
+		if isAdmin {
+			paths = rule.origSymbolicPaths
+		}
+		for _, p := range *paths {
+			if !p.Conn.TCPUDPSet().IsEmpty() {
+				policies.addNewPolicy(p, inbound, isAdmin, rule.origRule.Action)
+			}
+		}
+	}
+}
+
+func (policies *k8sPolicies) addNewPolicy(p *symbolicexpr.SymbolicPath, inbound, isAdmin bool, action dfw.RuleAction) {
+	srcSelector := toSelector(p.Src)
+	dstSelector := toSelector(p.Dst)
+	if isAdmin {
+		ports := connToAdminPolicyPort(p.Conn)
+		policies.addAdminNetworkPolicy(srcSelector, dstSelector, ports, inbound,
+			abstractToAdminRuleAction[action], fmt.Sprintf("(%s: (%s)", action, p.String()))
+	} else {
+		ports := connToPolicyPort(p.Conn)
+		policies.addNetworkPolicy(srcSelector, dstSelector, ports, inbound, p.String())
+	}
+}
+
+func (policies *k8sPolicies) addNetworkPolicy(srcSelector, dstSelector *meta.LabelSelector,
+	ports []networking.NetworkPolicyPort, inbound bool,
+	description string) {
+	pol := newNetworkPolicy(fmt.Sprintf("policy_%d", len(policies.networkPolicies)), description)
+	policies.networkPolicies = append(policies.networkPolicies, pol)
+	if inbound {
+		from := []networking.NetworkPolicyPeer{{PodSelector: srcSelector}}
+		rules := []networking.NetworkPolicyIngressRule{{From: from, Ports: ports}}
+		pol.Spec.Ingress = rules
+		pol.Spec.PolicyTypes = []networking.PolicyType{"Ingress"}
+		pol.Spec.PodSelector = *dstSelector
+	} else {
+		to := []networking.NetworkPolicyPeer{{PodSelector: dstSelector}}
+		rules := []networking.NetworkPolicyEgressRule{{To: to, Ports: ports}}
+		pol.Spec.Egress = rules
+		pol.Spec.PolicyTypes = []networking.PolicyType{"Egress"}
+		pol.Spec.PodSelector = *srcSelector
+	}
+}
+
+func (policies *k8sPolicies) addAdminNetworkPolicy(srcSelector, dstSelector *meta.LabelSelector,
+	ports []admin.AdminNetworkPolicyPort, inbound bool, action admin.AdminNetworkPolicyRuleAction, description string) {
+	pol := newAdminNetworkPolicy(fmt.Sprintf("admin_policy_%d", len(policies.adminNetworkPolicies)), description)
+	policies.adminNetworkPolicies = append(policies.adminNetworkPolicies, pol)
```

Comment:
can also add annotation with details of the original nsx rule ID ?

Reply:
done